### ISSUE TITLE
Allow reading only a subset of rows

### ISSUE THREAD

**Issue by davidw76** [crealytics/spark-excel#40]
This would be useful for the project I work on.
We first ingest only 1,000 rows (inferring the schema) and show the user a "preview" of the dataset.
The user can then override column types/names before ingesting the entire dataset.
If the dataset is large then it is much faster to be able to ingest only the first 1,000 records at the preview stage.

**Comment by nightscape** [comment]
Another option would be to add a virtual `rowNumber` column to the exported DataFrame and implement [`PrunedFilteredScan`](https://spark.apache.org/docs/2.2.0/api/scala/index.html#org.apache.spark.sql.sources.PrunedFilteredScan) in addition to `PrunedScan` in `ExcelRelation`.
Then you could do something like
```scala
val preview = existingExcelDf.where(col("rowNumber") >= 5 && col("rowNumber") <= 1005)
```
What I'm not sure about though is if you to have to handle all filters (meaning Spark will not do any post-filtering) when you implement `PrunedFilteredScan`.

**Comment by nightscape** [comment]
Ok, actually reading the documentation link I posted would have answered my question :stuck_out_tongue_winking_eye:
Spark will still do post-filtering, so it's OK to return false positives.
What do you think of implementing `PrunedFilteredScan` then and using that to do the start and end row logic?

**Comment by davidw76** [comment]
That sounds good. I will take a look at doing it that way.

**Comment by jaceklaskowski** [comment]
Is anyone working on it? My project would benefit from such a feature a lot.

**Comment by nightscape** [comment]
Just iterating over all issues again. Having this implemented should also fix #59.
@davidw76 @jaceklaskowski are you still interested in this?

**Comment by davidw76** [comment]
Yes, for us it's more important to be able to ingest the first n rows, but for others I guess skipping n rows is more useful. So it would be good to be able to specify a range.

**Comment by nightscape** [comment]
What I'm wondering is if we should handle the "skipping n rows because the header row is in line n+1" separately (see #65)?
Then the proposed solution with the artificial `row_number` column would still work and would allow additional features like

- returning the first 100 and the last 100 rows (the end of the file might be interesting, too)
- getting a column that uniquely identifies each row for free

OTOH, if all the majority of users need is a start and an end row then it would be overkill...

**Comment by nightscape** [comment]
I just released `0.11.0`. Please have a look at the corresponding `CHANGELOG` and the `README` about the new `dataAddress` option which supersedes `sheetName`, `startColumn`, `endColumn` and `skipFirstRows` and should hopefully cover this use-case as well.

**Comment by nightscape** [comment]
Check out `dataAddress`
